### PR TITLE
[Max Price] - Prevent buying scalper stuff

### PR DESCRIPTION
**Pull request by pr1ntr**
Just adds a reserve price to the amz config.

### CODE REVIEW

**Review by worst-mod-confirmed — CHANGES_REQUESTED** [review]
Fix line 195.

**Comment by DakkJaniels** [comment]
Suggestion: Instead of adding smile to url template, just remove www from the url template and that needs to be included if you are using the non-smile amazon site. As written, I think this would break amazon for non US users.

File: stores/amazon.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 AMAZON_URLS = {
     "BASE_URL": "https://www.{domain}/",
-    "CART_URL": "https://www.{domain}/gp/aws/cart/add.html",
+    "CART_URL": "https://smile.{domain}/gp/aws/cart/add.html",
```
Comment (DakkJaniels) [inline stores/amazon.py:21]: delete "smile."

File: amazon_config.template_json [hunk 2]
```diff
@@ -2,5 +2,6 @@
   "username": "",
   "password": "",
   "asin_list": ["B07JH53M4T","B08HR7SV3M"],
-  "amazon_website": "amazon.co.uk"
+  "amazon_website": "amazon.co.uk",
```
Comment (DakkJaniels) [inline amazon_config.template_json:5]: add "www." or "smile.". I don't think "smile.amazon.*" exists for all language versions of the amazon site.

File: README.md [hunk 3]
```diff
@@ -69,15 +69,18 @@ Thankfully Amazon.com has 1 click order canceling so its not a huge issue. We ar
...
 
-Make a copy of `amazon_config.template_json` to `amazon_config.json`:
+Make a copy of `amazon_config.template_json` to `amazon_config.json`:   
 ```json
 {
   "username": "",
   "password": "",
   "asin_list": ["B07JH53M4T","B08HR7SV3M"],
-  "amazon_website": "amazon.com"
+  "amazon_website": "amazon.com",
```
Comment (DakkJaniels) [inline README.md:78]:
```suggestion
  "amazon_website": "www.amazon.com",
```

File: stores/amazon.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
 AMAZON_URLS = {
     "BASE_URL": "https://www.{domain}/",
```
Comment (DakkJaniels) [inline stores/amazon.py:20]:
```suggestion
    "BASE_URL": "https://{domain}/",
```

File: README.md (hunk 3 above)
Comment (DakkJaniels) [inline README.md:78]:
```suggestion
  "amazon_website": "www.amazon.com",
```

File: amazon_config.template_json (hunk 2 above)
Comment (DakkJaniels) [inline amazon_config.template_json:5]:
```suggestion
  "amazon_website": "www.amazon.co.uk",
```

File: stores/amazon.py (hunk 1 above)
Comment (DakkJaniels) [inline stores/amazon.py:21]:
```suggestion
    "CART_URL": "https://{domain}/gp/aws/cart/add.html",
```

File: stores/amazon.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
...
-    "CART_URL": "https://www.{domain}/gp/aws/cart/add.html",
+    "CART_URL": "https://smile.{domain}/gp/aws/cart/add.html",
 }
 CHECKOUT_URL = "https://www.{domain}/gp/cart/desktop/go-to-checkout.html/ref=ox_sc_proceed?partialCheckoutCart=1&isToBeGiftWrappedBefore=0&proceedToRetailCheckout=Proceed+to+checkout&proceedToCheckout=1&cartInitiateId={cart_id}"
```
Comment (DakkJaniels) [inline stores/amazon.py:23]:
```suggestion
CHECKOUT_URL = "https://{domain}/gp/cart/desktop/go-to-checkout.html/ref=ox_sc_proceed?partialCheckoutCart=1&isToBeGiftWrappedBefore=0&proceedToRetailCheckout=Proceed+to+checkout&proceedToCheckout=1&cartInitiateId={cart_id}"
```

File: stores/amazon.py [hunk 6]
```diff
@@ -179,10 +184,18 @@ def something_in_stock(self):
...
-            log.info("One or more items in stock!")
+        price_element = self.driver.find_elements_by_xpath('//td[@class="price item-row"]')
+        if price_element:
+            price = float(price_element[0].text[1:])
+            log.info(f'Item Cost: {price}')
+            if price <= self.reserve:
+                log.info("One or more items in stock and under reserve!")
+                return True
+            else:
+                log.info("Shit costs too much fuck that")
```
Comment (DakkJaniels) [inline stores/amazon.py:195]:
```suggestion
                log.info("No stock available under reserve price")
```

**Review by DakkJaniels — CHANGES_REQUESTED** [review]
Modified to URLs to remove "www." to make it easier to switch to smile.amazon.*

File: stores/amazon.py [hunk 7]
```diff
@@ -65,6 +68,7 @@
 ]
 ORDER_COMPLETE_TITLES = [
     "Amazon.com Thanks You",
+    "AmazonSmile Thanks You"
```
Comment (DakkJaniels) [inline stores/amazon.py:71]:
```suggestion
    "AmazonSmile Thanks You",
```

**Review by worst-mod-confirmed — DISMISSED** [review]

**Comment by DakkJaniels** [comment]
Make sure the comma gets added before this is pulled in. https://github.com/Hari-Nagarajan/nvidia-bot/pull/275#discussion_r519065730

**Review by worst-mod-confirmed — APPROVED** [review]

File: stores/amazon.py [hunk 8]
```diff
@@ -179,10 +184,18 @@ def something_in_stock(self):
         f.set(params)
         self.driver.get(f.url)
         self.check_if_captcha(self.wait_for_pages, ADD_TO_CART_TITLES)
-        if self.driver.find_elements_by_xpath('//td[@class="price item-row"]'):
-            log.info("One or more items in stock!")
+        price_element = self.driver.find_elements_by_xpath('//td[@class="price item-row"]')
+        if price_element:
+            price = float(price_element[0].text[1:])
```
Comment (DakkJaniels) [inline stores/amazon.py:189]: Sorry, forgot to include this before. This fails if the price has a comma in it. Could use package that has locale, which would be more robust, but for US (or anywhere that uses comma for separating digit groupings):
```suggestion

            price_str = price_element[0].text[1:]
            price = float(str_price.replace(',',''))
```

**Comment by DakkJaniels** [comment]
I think this has been implemented in the master now.

**Pull request closed without merging**